full-service/src/service/sync.rs

```diff
@@ -311,100 +305,101 @@ fn sync_thread_entry_point(
-                Err(mc_ledger_db::Error::NotFound) => {
-                    return Ok(SyncAccountOk::NoMoreBlocks);
-                }
-                Err(err) => {
-                    return Err(err.into());
-                }
-            };
-
-            log::trace!(
-                logger,
-                "processing {} outputs and {} key images from block {} for account {}",
-                block_contents.outputs.len(),
-                block_contents.key_images.len(),
-                account.next_block,
-                account_id,
-            );
+            let block_index = account.next_block as u64;
+            if block_index > final_block_index {
+                // This account has been synced with all blocks currently in the ledger.
+                return Ok(AccountSyncStatus::Synced);
+            }
+            let block_contents = ledger_db.get_block_contents(account.next_block as u64)?;
 
-            // Match tx outs into UTXOs.
+            // Outputs in this block received by `account`?
             let output_txo_ids = process_txos(
-                &conn,
                 &block_contents.outputs,
+                block_index,
                 &account,
-                account.next_block,
+                &conn,
                 logger,
             )?;
 
-            // Note: Doing this here means we are updating key images multiple times, once
-            // per account. We do actually want to do it this way, because each account may
-            // need to process the same block at a different time, depending on when we add
-            // it to the DB.
+            // Key images spent in this block that correspond to prior TXOs owned by `account`.
+            // This also updates account.next_block in the DB.
             account.update_spent_and_increment_next_block(
                 account.next_block,
                 block_contents.key_images,
                 &conn,
             )?;
 
-            // Add a transaction for the received TXOs
+            // Include received TXOs in the TransactionLog.
             TransactionLog::log_received(
                 &output_txo_ids,
                 &account,
-                account.next_block as u64,
+                block_index,
                 &conn,
             )?;
-            Ok(SyncAccountOk::MoreBlocksPotentiallyAvailable)
+            Ok(AccountSyncStatus::MoreBlocksPotentiallyAvailable)
         })?;
-        // Early out of the loop if we hit NoMoreBlocks
-        if let SyncAccountOk::NoMoreBlocks = sync_status {
-            return Ok(SyncAccountOk::NoMoreBlocks);
+
+        // Early exit.
+        if sync_status == AccountSyncStatus::Synced {
+            return Ok(AccountSyncStatus::Synced);
         }
     }
-    Ok(SyncAccountOk::MoreBlocksPotentiallyAvailable)
+    Ok(AccountSyncStatus::MoreBlocksPotentiallyAvailable)
 }
 
-/// Helper function for matching a list of TxOuts to a given account.
+
+/// TODO: What does this do?
```

Comment:
Is there a short description of what the intention here is? I'm having a hard time describing this function accurately in one sentence.